staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/status_strategy.go

```diff
@@ -94,8 +100,17 @@ func (a statusStrategy) ValidateUpdate(ctx context.Context, obj, old runtime.Obj
 		return field.ErrorList{field.Invalid(field.NewPath(""), old, fmt.Sprintf("has type %T. Must be a pointer to an Unstructured type", old))}
 	}
 
+	var options []validation.ValidationOption
+	var celOptions []cel.Option
+	var correlatedObject *common.CorrelatedObject
+	if utilfeature.DefaultFeatureGate.Enabled(apiextensionsfeatures.CRDValidationRatcheting) {
+		correlatedObject = common.NewCorrelatedObject(uNew.Object, uOld.Object, &model.Structural{Structural: a.structuralSchema})
+		options = append(options, validation.WithRatcheting(correlatedObject.Key("status")))
+		celOptions = append(celOptions, cel.WithRatcheting(correlatedObject))
```

Comment:
it looks like we can encounter CRDs that said they wanted a status subresource but didn't define a status property, and this leads to a panic in that case

can we detect / guard against that in correlatedObject or here somehow?

xref https://github.com/kubernetes/kubernetes/pull/133721 https://github.com/kubernetes/kubernetes/issues/133651